### DISCUSSION TITLE
When does the run failure sensor trigger? Does it trigger immediately when any step of a job fails or only when the entire job completes?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18193] [Q&A] [answered]
sorry for being so support-needing today, but the`run_failure_sensor` does that throw when the complete job is failed, or when a step inside a job fails. I.e the job is two disjoint length of operations (A1->A2->A3 and B1->B2->B3), if A1 fails, but the B train continues, does the sensor the trigger "immeddiately" or only after the B train terminates?

<https://docs.dagster.io/concepts/partitions-schedules-sensors/sensors#run-failure-sensor> (would be nice if this was explicit in the docs. maybe i miss something)

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1700559550172269?thread_ts=1700559550.172269&cid=C01U954MEER).*

**Comment by benpankow** [comment, marked as answer]
The `run_failure_sensor ` sensor specifically monitors for `DagsterRunStatus.FAILURE`, which indicates that a run has concluded & no compute is active. 

This means it will not trigger until all steps in a run have completed. In the above example, the sensor triggers only after the `B` steps have finished.
